ts/src/hyperliquid.ts

```diff
@@ -548,6 +559,17 @@ export default class hyperliquid extends Exchange {
         const taker = this.safeNumber (fees, 'taker');
         const maker = this.safeNumber (fees, 'maker');
         const amountPrecision = this.safeInteger (market, 'szDecimals');
+        const price = this.safeNumber (market, 'markPx', 0);
+        const significantDigits = 5;  // Remove this line after network upgrade
+        // const significantDigits = Math.max (5, price !== 0 ? Math.floor (Math.log10 (price)) + 1 : 1);  // Uncomment this line after network upgrade
+        let pricePrecision = 0;
+        if (price > 0 && price < 1) {
+            pricePrecision = Math.abs (Math.log10 (price)) + significantDigits; // 5 significant digits
+            pricePrecision = Math.min (6 - amountPrecision, Math.floor (pricePrecision)); // 6 max decimals
+        } else {
+            const integerDigits = price === 0 ? 1 : Math.floor (Math.log10 (price)) + 1;
+            pricePrecision = Math.min (6 - amountPrecision, significantDigits - integerDigits); // 6 max decimals, 5 significant digits
```

Comment:
this logic seems duplicated, maybe we can dry it a bit by abstracting it into a method that receives the max decimals and the price no?

Reply:
yes, this is the same in `fetchSpotMarkets()` and `parseMarket()` I wasn't sure how to fix that. In python I think it's not good practice to define a method that doesn't exist in the base class.

So is it ok to define a new function, or is there a 'dummy' function in the base class we can kind of 'abuse'..

Reply:
@gaardiolor you can define a helper function within the class with no issues at all, we do it frequently. 

(all user facing methods should also be declared in exchange.ts but this is different)

Reply:
ok, I created the helper function